nixos/tests/teleport.nix

```diff
@@ -0,0 +1,86 @@
+import ./make-test-python.nix ({ pkgs, ... }: {
+  name = "teleport";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ ymatsiuk ];
+  };
+
+  nodes = {
+    client = { config, pkgs, ... }: {
+      services.teleport = {
+        enable = true;
+        settings = {
+          teleport = {
+            nodename = "client";
+            advertise_ip = "192.168.1.20";
+            auth_token = "8d1957b2-2ded-40e6-8297-d48156a898a9";
+            auth_servers = [ "192.168.1.10:3025" ];
+          };
+          ssh_service.enabled = true;
+          proxy_service.enabled = false;
+          auth_service.enabled = false;
+        };
+        debug.enable = true;
+      };
+      networking.interfaces.eth1.ipv4.addresses = [{
+        address = "192.168.1.20";
+        prefixLength = 24;
+      }];
+    };
+    server = { config, pkgs, ... }: {
+      services.teleport = {
+        enable = true;
+        settings = {
+          teleport = {
+            nodename = "server";
+            advertise_ip = "192.168.1.10";
+          };
+          ssh_service.enabled = true;
+          proxy_service.enabled = true;
+          auth_service = {
+            enabled = true;
+            tokens = [ "node:8d1957b2-2ded-40e6-8297-d48156a898a9" ];
+          };
+        };
+        diag.enable = true;
+        extraOpts = [
+          "--labels=RandomLabelKey=RandomLabelValue"
+        ];
+      };
+      environment.systemPackages = [ pkgs.jq ];
+      networking.firewall.allowedTCPPorts = [ 3025 ];
+      networking.interfaces.eth1.ipv4.addresses = [{
+        address = "192.168.1.10";
+        prefixLength = 24;
+      }];
+    };
+    standalone = { config, pkgs, ... }: {
+      services.teleport.enable = true;
+    };
+  };
+
+  testScript = ''
+    with subtest("start client after server is inititalized and accepting connections"):
+        server.wait_for_open_port("3025")
+        client.wait_for_open_port("3022")
+
+    with subtest("extraOpts, diag and debug works"):
+        server.succeed("tctl get nodes --format=json | jq -e '.[] | select(.spec.hostname==\"server\") | .metadata.labels.RandomLabelKey==\"RandomLabelValue\"'")
+        server.wait_for_open_port("3000")
+        client.succeed("journalctl -u teleport.service --grep='DEBU'")
+
+    with subtest("client joins"):
+        server.wait_until_succeeds("tctl get nodes --format=json | jq -e '.[] | select(.spec.hostname==\"client\") | .spec.hostname==\"client\"'")
+
+    server.shutdown()
+    client.shutdown()
+
+    with subtest("teleport starts without the config"):
+        standalone.wait_for_open_port("3025")
+        standalone.wait_for_open_port("3080")
+        standalone.wait_for_open_port("3022")
+
+    standalone.shutdown()
```

Comment:
This test is fully disconnected from the "main" test, maybe it could be two seperate tests in the same file, this parallelizes better

Reply:
I'll try a bit different approach here and if that doesn't work, consider separating test, ok?